pydantic_ai_slim/pydantic_ai/durable_exec/temporal/_model.py

```diff
@@ -75,28 +78,35 @@ def __init__(
         deps_type: type[AgentDepsT],
         run_context_type: type[TemporalRunContext[AgentDepsT]] = TemporalRunContext[AgentDepsT],
         event_stream_handler: EventStreamHandler[Any] | None = None,
+        model_instances: Mapping[str, Model] | None = None,
+        provider_factory: TemporalProviderFactory | None = None,
     ):
         super().__init__(model)
         self.activity_config = activity_config
         self.run_context_type = run_context_type
         self.event_stream_handler = event_stream_handler
+        self._model_id_var: ContextVar[str | None] = ContextVar('_temporal_model_id', default=None)
+        self._model_instances = dict(model_instances or {})
+        self._provider_factory = provider_factory
 
         @activity.defn(name=f'{activity_name_prefix}__model_request')
-        async def request_activity(params: _RequestParams) -> ModelResponse:
-            return await self.wrapped.request(
+        async def request_activity(params: _RequestParams, deps: Any) -> ModelResponse:
```

Comment:
`deps` needs to be optional here so that we don't break on activity payloads created in the past.